src/components/ProtectedRoute.tsx

```diff
@@ -24,7 +24,7 @@ export function ProtectedRoute({ children }: { children: JSX.Element }) {
     return <ErrorBlockerMessage message="Session expired. Please log in again. ⚠️" locationMessage="Go to Login" />;
   }
 
-  if (loading || !user) {
+  if (loading || !user || !profileSetupComplete && location.pathname !== '/setup') {
```

Comment:
This will remove the "glitch" where users would see the dashboard before being redirected to `/setup`.
So I protect to not allow showing it.